lib/ui/home/home_view_model.dart

```diff
@@ -110,6 +118,20 @@ class HomeViewModel extends StateNotifier<HomeViewState> {
     }
   }
 
+  Future _getCachedSurveys() async {
+    final result = await _getCachedSurveysUseCase.call();
+
+    if (result is Failed<List<SurveyModel>>) {
+      final error = result.getErrorMessage();
+      state = HomeViewState.error(error);
+    } else {
+      final surveys = (result as Success<List<SurveyModel>>).value;
+      final surveyUiModels =
+          surveys.map((survey) => SurveyUiModel.fromModel(survey));
+      _surveysStreamController.add(surveyUiModels.toList(growable: false));
+    }
+  }
```

Comment:
@hoangnguyen92dn those pieces of code are duplicated from `getSurveys`. Can you extract the duplicated part to another method to reuse between the methods `getSurveys` and `_getCachedSurveys`